File: model/gallery.go
package model

import (
	"time"

	"github.com/allape/gocrud"
	"github.com/allape/gophorward"
)

const MaxGalleryNameLength = 50

type Gallery struct {
	gocrud.Base
	Name        string            `json:"name"`
	Keywords    string            `json:"keywords"`
	IsPublic    bool              `json:"isPublic"`
	Description string            `json:"description"`
	CreatedBy   gophorward.UserID `json:"createdBy"`
	Enabled     bool              `json:"enabled"`
}

type GalleryItem struct {
	GalleryID gocrud.ID `json:"galleryId" gorm:"primaryKey"`
	ItemID    gocrud.ID `json:"itemId" gorm:"primaryKey"`
	CreatedAt time.Time `json:"createdAt" gorm:"autoCreateTime;<-:create"`
}

type GalleryTag struct {
	GalleryID gocrud.ID `json:"galleryId" gorm:"primaryKey"`
	TagID     gocrud.ID `json:"tagId" gorm:"primaryKey"`
	CreatedAt time.Time `json:"createdAt" gorm:"autoCreateTime;<-:create"`
}

File: ui/common/src/model/gallery.ts
import { IBase } from "@allape/gocrud";
import { IItem } from "./item.ts";
import { ITag } from "./tag.ts";

export interface IGallery extends IBase {
  name: string;
  keywords: string;
  isPublic: boolean;
  description: string;
  createdBy: string;
  enabled: boolean;
}

export interface IGalleryItem extends Pick<IBase, "createdAt"> {
  galleryId: IGallery["id"];
  itemId: IItem["id"];
}

export interface IGalleryTag extends Pick<IBase, "createdAt"> {
  galleryId: IGallery["id"];
  tagId: ITag["id"];
}
